rfd/0087-access-request-notification-routing.md

```diff
@@ -0,0 +1,207 @@
+in the `expression` field. The predicate evaluation in `expression` field is expected to return a pair containing the plugin
+name and the recipient.b If one or both fields are empty, the target should not be set.
+
+If `expression` is set, `condition`, `recipient` and `plugin` should not be used. Teleport must validate
+this on resource creation and fail immediately instead of silently ignoring fields.
+
+#### Combining with suggested_reviewers
+
+The safest way would be for this new feature to not interact with suggested reviewers and let the plugins deal with it.
+
+Using `suggested_reviwer` implies the user knows which plugin will be used to send the access request. 
+The string has to make sense and be allowed by the plugin. Today most plugins support email addresses
+and reject other recipients, but somes are just ignoring the field (pagerduty). We might want to clarify
+what can be entered in this field and how different plugins are reacting to it so user has insights on
+what to type and what to expect.
+
+Using `suggested_reviewer` with two plugins supporting it (e.g. Slack and MsTeams, or two Slack plugins on different workspaces)
+can lead to the suggested reviewer being notified multiple times. One way to mitigate this would be to
+offer a plugin configuration flag to ignore suggested reviewers (e.g. suggested reviewers are only
+pinged on Slack).
+
+#### JIT resource access request
+
+While `access_rouring_rule` should work with both role access requests and resource access requests,
+using it withresource access request can cause some friction for the admins writing the rules:
+
+Only resource IDs are available through access requests: writing a rule to filter based on requested resources
+requires knowing resources IDs. Rules with lists of IDs can be hard to read and maintain. For resource access requests routing,
+admins would likely prefer to route the requests based on resources labels/tags.
+
+A workaround would be to resolve the resources (either before evaluating the rule or through a new predicate function).
+
+This is currently out of the scope of this RFD and could be implemented if the feature gets user traction.
+
+### Plugin considerations
+
+This change must be backward compatible:
+- old plugins should ignore `targets` and get the recipients from the `role_to_recipient` map (or annotation for pagerduty);
+- each plugin should expose an `honor_suggested_reviwers` configuration flag, true by default;
+- each plugin should have a default name and listen to it by default in the `targets`, this name should be the same used to store plugin data;
+- admins can configure the plugin name through its configuration, names should always be prefixed by the plugin type (e.g. `slack-teanA`,
+  `msteams-teamB`) to limit conflicts between two plugins with the same names;
+- in DEBUG mode, each plugin should log if it is ignoring a target (because plugin name does not match for example) to ease troubleshooting; and
```

Comment:
Nit: Just an implementation details but I'd probably just log at as Info. Should not clutter logs too much and it is a useful info to have imo, to be able to see whether what the plugin does with each request.